### DISCUSSION TITLE
SPI6 is not working

### DISCUSSION THREAD

**Discussion by mistermaeks** [Pi4J/pi4j#419] [Q&A]
Hi there,
i have a problem that i cannot use SPI1 because the mode 1 ( CPHA=1) is not possible as SPI1 is just a SPI mini. I try to communicate with an ADS1118 A/D Changer. As my board required due to design the GPIO17-21 i was trying to use the SPI6 instead which is on the CM4 module a fully SPI. With python i already managed to get the values from the ADS1118. The problem is that with pigpio-spi the bus number should not exceed 1, so only SPI0 & SPI1 is supported. With the raspberrypi-spi it is allowd but my device is not opening
`Spi to ADS1118-BUS_6-CS_0 open: false`.
Is it foreseen to support in pigpio-spi due to the board model also the other Spi buses or is there a solution for the rasperrypi-spi?
Thanks a lot in advance for pushing me in the right direction.

**Comment by FDelporte** [comment]
I see there are 6 busses defined here:
https://github.com/Pi4J/pi4j-v2/blob/develop/pi4j-core/src/main/java/com/pi4j/io/spi/SpiBus.java

And indeed in the PiGpio-implementation only 0 and 1 are supported:
https://github.com/Pi4J/pi4j-v2/blob/develop/plugins/pi4j-plugin-pigpio/src/main/java/com/pi4j/plugin/pigpio/provider/spi/PiGpioSpi.java#L106

This may get fixed if we can add SPI support to one of the other providers.
There is already a pull request for RPi 5 here https://github.com/Pi4J/pi4j-v2/pull/390
And @taartspi is also looking into this issue.

Sorry, I don't know how to fix this at this moment, but if you can share more details of your code and maybe point us to a possible implementation to solve this, things can move faster...

**Reply by mistermaeks** [reply]
Hi,
is there a specific reason why it is only for spi bus 0 & 1 rather than downwards compatibitlity to the other raspi hardware?
The code could check also the boardmodel if any thing else than unknown we know how many buses are supported by the hardware. So the need for prohibit everything above 1 would be not longer given. Or is there another reason why it is not supporting spi6.

**Reply by FDelporte** [reply]
I checked a bit further and the library implementation for PiGpio also indeed mentions 0 or 1:
https://github.com/Pi4J/pi4j-v2/blob/develop/libraries/pi4j-library-pigpio/src/main/java/com/pi4j/library/pigpio/PiGpio_SPI.java#L59

So a solution with PiGpio will not be available and another provider will be needed to provide SPI, hopefully making all 6 accessible.
